### PR TITLE
Otel deployment

### PR DESCRIPTION
**Pull request by longrunningprocess**

### CODE REVIEW

**Review by myieye — COMMENTED** [review]
Looks good to me. The only thing that I would change is renaming `config.yaml` to `collector-config.yaml`.

File: otel/config.yaml
[the remark this answers is not in the document]
Comment (longrunningprocess) [reply otel/config.yaml:1]: gotcha:  fixed in [pr feedback](https://github.com/sillsdev/languageforge-lexbox/commit/b713b9856e556e7b710f48661c5e341e579cbed1)

File: deployment/otel-collector-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      containers:
+      - name: otel-collector
+        image: ghcr.io/sillsdev/lexbox-otel-collector
+        # https://kubernetes.io/docs/concepts/configuration/manage-resources-containers
+        resources:
+          requests: # TODO: need to establish resource limits, possibly after seeing it in action for some regular and/or load testing
+            memory: 260Mi
+            cpu: 0.1
+          limits:
+            memory: 400Mi
```
[the remark this answers is not in the document]
Comment (longrunningprocess) [reply deployment/otel-collector-deployment.yaml:44]: I did not notice it, lucky coincidence :-)

File: deployment/otel-collector-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        app: otel-collector
+    spec:
+      containers:
+      - name: otel-collector
+        image: ghcr.io/sillsdev/lexbox-otel-collector
+        # https://kubernetes.io/docs/concepts/configuration/manage-resources-containers
+        resources:
+          requests: # TODO: need to establish resource limits, possibly after seeing it in action for some regular and/or load testing
+            memory: 260Mi
+            cpu: 0.1
```
[the remark this answers is not in the document]
Comment (longrunningprocess) [reply deployment/otel-collector-deployment.yaml:42]: pretty much but here's some additional detail:  https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu

**Comment by hahn-kev** [comment]
so am I right in understanding that if we want to change the configuration of the collector we need to rebuild and deploy?

**Comment by hahn-kev** [comment]
Partiality I was confirming, but I'm not sure how I feel about requiring a rebuild for a simple config change like that, are we ok with being unable to change the config if we can't make a new build right now for some reason.

Something else I thought of more recently is that we do need to have some config specific to deployment. For example we probably want to tag the telemetry based on the environment (staging or prod), so this is something that I'd like to have accounted for. I'm not sure if that would be defined in this config, or if it needs to be on the application side, but it would certainly be simpler to do it in one place here than in each application.

**Comment by longrunningprocess** [comment]
> Partiality I was confirming, but I'm not sure how I feel about requiring a rebuild for a simple config change like that, are we ok with being unable to change the config if we can't make a new build right now for some reason.

this depends on the circumstances of course, if it's a temp change for troubleshooting purposes and the app reads that config regularly, we can do that out in the container.  If it's a permanent change going forward, it belongs in the image so no matter where you use it, it will be the same.

> Something else I thought of more recently is that we do need to have some config specific to deployment. For example we probably want to tag the telemetry based on the environment (staging or prod), so this is something that I'd like to have accounted for. I'm not sure if that would be defined in this config, or if it needs to be on the application side, but it would certainly be simpler to do it in one place here than in each application.

Typically, we'd control something like this via env vars, e.g., `environment=[prod|dev]`, but it all depends on what we're controlling and how... it might be something that is derived from the host of the requestor... we just need to understand all of that better to make a solid determination

**Pull request merged**